### ISSUE TITLE
p2_c1_2022

### ISSUE THREAD

**Issue by aguegea** [POO-ITBA/2024_02#20]
Buenas tengo una consulta sobre el ejercicio 1 del parcial: [https://docs.google.com/document/d/1-kn36oOuQyditiZtrDk2S6v3-wULrmTKM9uUHbsS3k4/edit?tab=t.0](url)
 
En la solución dada la función toBag devuelve una instancia de HashMap, pero en el test hay una parte que pide devolver el "mayor" y el "menor" elemento de bag. No termino de comprender cual sería el criterio para decir que elemento es el mayor y el menor en un mapa sin orden.

Yo lo que hice es que la función toBag devuelva una instancia de TreeMap, y para el mayor y menor utilicé firstKey y lastKey. Lo que me quedaba diferente con la solución es que mi tipo genérico T debía implementar Comparable<T> para que el TreeMap pueda ordenarlo.

**Comment by fmeola** [comment]
Hola @aguegea 
Me parece que consultaste la solución del parcial de ese cuatrimestre y no la del recuperatorio. Lo que planteás del TreeMap es correcto.

Para el parcial
https://docs.google.com/document/d/13jVmiCLxFOp69UdFTgBp2F5Y852AjJSsVfV8OFs0LJo/edit?tab=t.0
se pedía un bag sin orden, por eso HashMap
https://docs.google.com/document/d/14cNoyIus42qBuHecYKAqRnPaX16hUt5TD_2gh43UDzw/edit?tab=t.0

Para el recuperatorio
https://docs.google.com/document/d/1-kn36oOuQyditiZtrDk2S6v3-wULrmTKM9uUHbsS3k4/edit?tab=t.0
se pedía un bag con orden y métodos para conseguir el menor o mayor elemento de la bolsa, por eso TreeMap con el E extends Comparable y los métodos firstKey y lastKey
https://docs.google.com/document/d/1PpCQVcCplZBMiSb9jlM7TtcL7L05pYL9r_uAMKy5PHo/edit?tab=t.0

**Comment by aguegea** [comment]
Ahh me confundí de parcial. Donde puedo ver las soluciones de los recuperatorios? En la carpeta de segundos parciales aparecen sólo la de los parciales.

**Comment by fmeola** [comment]
No, fijate que en el mismo elemento de Campus tenés todos los enunciados y soluciones de segundos parciales y más abajo están todos los enunciados y soluciones de los recuperatorios de segundos parciales.
![Screenshot 2024-10-28 at 12 42 59](https://github.com/user-attachments/assets/76f9ff61-c851-4286-a89d-73b2eb0774ed)
